packages/build/tests/plugins/tests.js

```diff
@@ -1046,6 +1046,22 @@ test('Validate --node-path', async (t) => {
   await runFixture(t, 'node_version_simple', { flags: { nodePath: '/doesNotExist' } })
 })
 
+test('Provided --node-path version produces a node version warning if the build is being run in buildbot with local plugins', async (t) => {
+  const nodePath = getNodePath('9.0.0')
+  await runFixture(t, 'engines', { flags: { nodePath, mode: 'buildbot' } })
+})
+
+test('Provided --node-path version does not produce a node version warning if the build is being run locally', async (t) => {
+  const nodePath = getNodePath('9.0.0')
+  await runFixture(t, 'engines', { flags: { nodePath, mode: 'require' } })
```

Comment:
`mode: 'cli'` might be more useful since almost no users is using `mode: 'require'`.

Reply:
I keep mixing `cli` with the build's cli execution 😅

Reply:
Just for clarification for anyone else reading this: 
  - `mode: 'cli'` is the `netlify build` CLI command 
  - `mode: 'require'` is when doing `require('@netlify/build')` programmatically